src/main/scala/sparkz/core/network/peer/PeerManager.scala

```diff
@@ -137,42 +143,63 @@ object PeerManager {
     case class SeenPeers(howMany: Int) extends GetPeers[Seq[PeerInfo]] {
 
       override def choose(knownPeers: Map[InetSocketAddress, PeerInfo],
+                          peers: Map[InetSocketAddress, PeerInfo],
                           blacklistedPeers: Seq[InetAddress],
                           sparkzContext: SparkzContext): Seq[PeerInfo] = {
-        val recentlySeenNonBlacklisted = knownPeers.values.toSeq
+        val recentlySeenKnownPeers = knownPeers.values.toSeq
+          .filter { p =>
+            (p.connectionType.isDefined || p.lastHandshake > 0)
+          }
+        val recentlySeenNonBlacklisted = peers.values.toSeq
           .filter { p =>
             (p.connectionType.isDefined || p.lastHandshake > 0) &&
               !blacklistedPeers.exists(ip => p.peerSpec.declaredAddress.exists(_.getAddress == ip))
           }
-        Random.shuffle(recentlySeenNonBlacklisted).take(howMany)
+        Random.shuffle(recentlySeenKnownPeers ++ recentlySeenNonBlacklisted).take(howMany)
```

Comment:
Please correct me if I'm wrong, but does it means that knownPeers (aka configured peers) are immune to blacklisting. Is it intended?

Same for RandomPeerForConnectionExcluding

Reply:
Yes, known peers are meant for trusted peers such as forgers, and we don't want to blacklist them. If you think this is an issue, we can definitely discuss it 🙂.